test/zauction.test.ts

```diff
@@ -344,4 +334,43 @@ describe("zAuction Contract Tests", () => {
 
     await expect(tx).to.be.revertedWith("zAuction: data already consumed");
   });
+  it("Calculates minter royalty correctly", async () => {
+    // Royalty is set to 10% above
+    // Each WILD is 10^18, Bid is 15 WILD
+    const bid = "15000000000000000000";
+    const id = "12345";
+    const royalty = await zAuction.calculateMinterRoyalty(bid, id);
+    const decimal = royalty.toString();
+
+    // 10% of bid
+    expect(decimal).to.equal("1500000000000000000");
+  });
+  it("Calculates root owner royalty correctly", async () => {
+    // Royalty is set to 10% above
+    // Each WILD is 10^18, Bid is 15 WILD
+
+    const bid = "15000000000000000000";
+    const id = "123245";
+    await zAuction.setRootRoyaltyAmount(id, 10);
+    const royalty = await zAuction.calculateRootOwnerRoyalty(id, bid, id);
+    const decimal = royalty.toString();
+
+    // 10% of bid (one less zero)
+    expect(decimal).to.equal("1500000000000000000");
+  });
+  it("Gets the root parent of a domain that is already the root", async () => {
```

Comment:
Consider adding another test for non-root domains when testing the `rootDomainIdOf` function.

You can do something like this
```
mockRegistrar.parentof.whenCalledWith(12345).returns(123);
mockRegistrar.parentof.whenCalledWith(123).returns(1);
mockRegistrar.parentof.whenCalledWith(1).returns(0);
```

This way your `mockRegistrar` functions more like what you would expect in the real world.